pageserver/src/tenant/storage_layer/inmemory_layer.rs

```diff
@@ -89,6 +90,11 @@ pub struct InMemoryLayerInner {
 
     resource_units: GlobalResourceUnits,
 }
+pub(crate) struct InMemoryLayerIndexValue {
+    pub(crate) pos: u32,
+    pub(crate) len: u32,
+    pub(crate) will_init: bool, // XXX this blows up the size, can we shrink down `len`?
+}
```

Comment:
This is replacing the `u64` that we had before.

The `pos` and `len` definitely have a few bits left in practice (our values aren't that large and also `pos` doesn't need to go up to 4Gi because we don't allow such large in-memory layers). But I don't think there's a systemic limit in place at this time.

Wondering if I should implement such a systemic limit in a preliminary PR, then rely on that limit in this PR to stuff these 3 values into a single bitmasked u64.

Reply:
Yes but InMemory layer index caused OOMs in the past. Are our current controls good enough to prevent the OOMs?

Reply:
Done in https://github.com/neondatabase/neon/pull/8537/commits/e4af31cd7db04da1775fcc4ce6f74dc5b2582c3e

I don't think it's worth backporting / extracting as a preliminary PR